lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
We can tweak the diagnostic too to make that more clear, now that we're planning on having one compiler with two language modes.